shell/platform/common/cpp/client_wrapper/include/flutter/event_sink.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_EVENT_SINK_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_EVENT_SINK_H_
+
+namespace flutter {
+
+// Event callback. Events to be sent to Flutter application
+// act as clients of this interface for sending events.
+template <typename T>
+class EventSink {
+ public:
+  EventSink() = default;
+  virtual ~EventSink() = default;
+
+  // Prevent copying.
+  EventSink(EventSink const&) = delete;
+  EventSink& operator=(EventSink const&) = delete;
+
+  // Consumes end of stream. Ensuing calls to Success() or
+  // Error(), if any, are ignored.
+  void EndOfStream() { EndOfStreamInternal(); }
```

Comment:
Nit: it seems like it would be clear to make this the last method, rather than the first.